go/pod/value.go

```diff
@@ -0,0 +1,70 @@
+package pod
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/iden3/go-iden3-crypto/v2/poseidon"
+)
+
+// Currently, only supports string, boolean, and int.
+type PodValue struct {
+	kind    string
+	strVal  string
+	boolVal bool
+	intVal  int64
+}
+
+func (p *PodValue) UnmarshalJSON(data []byte) error {
+	var raw map[string]json.RawMessage
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+	if len(raw) != 1 {
+		return fmt.Errorf("invalid PodValue: must have exactly one key")
+	}
+	for k, v := range raw {
+		switch k {
+		case "string":
+			p.kind = "string"
+			return json.Unmarshal(v, &p.strVal)
+		case "boolean":
+			p.kind = "boolean"
+			return json.Unmarshal(v, &p.boolVal)
+		case "int":
+			p.kind = "int"
+			return json.Unmarshal(v, &p.intVal)
+		default:
+			return fmt.Errorf("unknown key %q in PodValue", k)
+		}
+	}
+	return nil
+}
+
+func (p PodValue) MarshalJSON() ([]byte, error) {
+	switch p.kind {
+	case "string":
+		return json.Marshal(map[string]string{"string": p.strVal})
+	case "boolean":
+		return json.Marshal(map[string]bool{"boolean": p.boolVal})
+	case "int":
+		return json.Marshal(map[string]int64{"int": p.intVal})
```

Comment:
This will also produce incorrect results for values > 53 bits. You can test this by hard-coding the `visitorCount` to `9007199254740992`, which produces a POD that Zupass isn't able to parse.

Reply:
resolved now -- see first test case for TestPodMarshal in `pod_test.go`